runtime/queries/cpp/highlights.scm

```diff
@@ -12,56 +10,90 @@
 (template_method
   name: (field_identifier) @function)
 
-(template_function
-  name: (identifier) @function)
-
 (function_declarator
   declarator: (qualified_identifier
     name: (identifier) @function))
 
 (function_declarator
   declarator: (qualified_identifier
-    name: (identifier) @function))
+    name: (qualified_identifier
+      name: (identifier) @function)))
 
 (function_declarator
   declarator: (field_identifier) @function)
 
 ; Types
 
-((namespace_identifier) @type
- (#match? @type "^[A-Z]"))
+(namespace_definition name: (identifier) @namespace)
+(using_declaration (identifier) @namespace)
+(namespace_identifier) @namespace
 
-(auto) @type
+(qualified_identifier name: (identifier) @type.enum.variant)
 
 ; Constants
 
 (this) @variable.builtin
-(nullptr) @constant
+(nullptr) @constant.builtin
 
 ; Keywords
 
-"catch" @keyword
-"class" @keyword
-"constexpr" @keyword
-"delete" @keyword
-"explicit" @keyword
-"final" @keyword
-"friend" @keyword
-"mutable" @keyword
-"namespace" @keyword
-"noexcept" @keyword
-"new" @keyword
-"override" @keyword
-"private" @keyword
-"protected" @keyword
-"public" @keyword
-"template" @keyword
-"throw" @keyword
-"try" @keyword
-"typename" @keyword
-"using" @keyword
-"virtual" @keyword
+[
+  "catch"
+  "co_await"
+  "co_return"
+  "co_yield"
+  "concept"
+  "consteval"
+  "constinit"
+  "delete"
+  "final"
+  "noexcept"
+  "new"
+  "requires"
+  "throw"
+  "try"
+  "typename"
+  "using"
+] @keyword
+
+"<=>" @operator
+
```

Comment:
would you mind adding template brackets? Distinguishing these from operators can be nice for readability.

```
(template_argument_list (["<" ">"] @punctuation.bracket))
(template_parameter_list (["<" ">"] @punctuation.bracket))
```